src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -249,8 +249,15 @@ internal SourceText TryReadFileContent(CommandLineSourceFile file, IList<Diagnos
             var filePath = file.Path;
             try
             {
-                using (var data = OpenFileForReadWithSmallBufferOptimization(filePath))
+                if (file.IsInputRedirected)
                 {
+                    using var data = Console.OpenStandardInput(bufferSize: 1);
+                    normalizedFilePath = filePath;
+                    return EncodedStringText.Create(data, Arguments.Encoding, Arguments.ChecksumAlgorithm, canBeEmbedded: EmbeddedSourcePaths.Contains(file.Path));
```

Comment:
>EmbeddedSourcePaths.Contains(file.Path) [](start = 122, length = 39)

Is this ever `true`? If not, consider replacing with `false`.

Reply:
It can be true, as we can compile N files and source for one of them comes from stdin, (which is referred as `stdin.cs`).